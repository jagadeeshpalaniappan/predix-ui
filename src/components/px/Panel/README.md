The `<Panel/>` component provides a container for placing controls or content. Can be fixed or relatively positioned, and persistent or shown and hidden.


```hint
Work in progress
```

## Usage

```react
const {Panel} = PxReact;
<Panel>
  Panel content goes here.
</Panel>
```


## Properties

```table
span: 6
rows:
  - Name: children
    Type: node
    Description: This contents of the chip.
  - Name: showIcon
    Type: bool
    Description: Toggle the icon on or off.
```


### Styling
The following custom properties are available for styling:

```code
rows:
  - Property: --px-chip-background-color
    Description: Background color for chip

  - Property: --px-chip-background-color--hover
    Description: Background color for chip on hover

  - Property: --px-chip-background-color--selected
    Description: Background color for chip on selected

  - Property: --px-chip-border-color
    Description: Border or outline color for chip

  - Property: --px-chip-border-color--hover
    Description: Border or outline color for chip on hover

  - Property: --px-chip-border-color--selected
    Description: Border or outline color for chip on selected

  - Property: --px-chip-icon-color
    Description: Icon color

  - Property: --px-chip-icon-color--hover
    Description: Icon color on hover

  - Property: --px-chip-text-color
    Description: Text color for chip content

  - Property: --px-chip-text-color--hover
    Description: Text color for chip content on hover

  - Property: --px-chip-text-color--selected
    Description: Text color for chip content on selected

  - Property: --px-chip-max-width
    Description: Max width of the chip (default: 120px)
```
