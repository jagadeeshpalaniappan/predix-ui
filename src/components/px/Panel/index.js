import React from 'react';
import classnames from 'classnames';

import styled, {css} from 'styled-components';

const Panel = styled.div`
  overflow:hidden;
  display: block;
  position: absolute;
  z-index: var(--px-panel-z-index, 100);
  background: var(--px-panel-bg-color--light, #FFF);
  box-shadow: -1px -1px 3px rgba(0,0,0,0.2),
               1px  1px 3px rgba(0,0,0,0.2);
  
  
  
 
  ${props => props.light && css`
    position: fixed;
  `}
 
  ${props => props.fixed && css`
    position: fixed;
  `}
 
  ${props => props.minimizable && css`
    
  `}

  ${props => props.persistent && css`
    position: fixed;
  `}
  
  ${props => ( props.position === 'left' || props.position === 'right' ) && css`
    height: auto;
    width: 0;
    top: 0;
    bottom: 0;
    transition: width 0.4s cubic-bezier(.78,.13,.16,.87);

    ${props => props.opened && css`
      width: var(--px-panel-size, 320px);
      overflow-y: auto;
    `}

  `}
  

   ${props => props.position === 'left'  && css`
    left: 0;
    ${props => props.fullSize && css`
      left: var(--px-panel-offset--left, 2rem);
      top: var(--px-panel-offset--top, 2rem);
      bottom: var(--px-panel-offset--bottom, 2rem);
    `} 
  `}
  

  ${props => props.position === 'right'  && css`
    right: 0;
    ${props => props.fullSize && css`
      right: var(--px-panel-offset--right, 2rem);
      top: var(--px-panel-offset--top, 2rem);
      bottom: var(--px-panel-offset--bottom, 2rem);
    `} 
  `}

  ${props => ( props.position === 'top' || props.position === 'bottom' ) && css`
    width: auto;
    height: 0;
    right: 0;
    left: 0;
    transition: height 0.4s cubic-bezier(.78,.13,.16,.87);
    ${props => props.opened && css`
      height: var(--px-panel-size, 320px);
      overflow-y: auto;
    `}
  `}

  
  ${props => props.position === 'top'  && css`
    top: 0;
    border-bottom: 1px solid var(--px-panel-border-color, gray);
    ${props => props.fullSize && css`
      right: var(--px-panel-offset--right, 2rem);
      left: var(--px-panel-offset--left, 2rem);
      top: var(--px-panel-offset--bottom, 2rem);
    `} 
  `}
  
  ${props => props.position === 'bottom'  && css`
    bottom: 0;
    border-top: 1px solid var(--px-panel-border-color, gray);
    ${props => props.fullSize && css`
      right: var(--px-panel-offset--right, 2rem);
      left: var(--px-panel-offset--left, 2rem);
      bottom: var(--px-panel-offset--bottom, 2rem);
    `} 
  `}

 
  
    
`;

const PanelContent = styled.div`
  display: block;
  ${props => props.opened && css`
    display: block;
    height: 100%;
  `}
`;

/**
 * Panel component
 */
export default ({
  style,
  

  position,
  fullSize,
  opened,
  fixed,
  persistent,
  floating,
  minimizable,
  children
}) => {
  return (
    <Panel 
      position={position} 
      minimizable={minimizable} 
      fullSize={fullSize} 
      floating={floating} 
      opened={opened} 
      fixed={fixed} >
      <PanelContent>{children}</PanelContent>
    </Panel>

  );
}
