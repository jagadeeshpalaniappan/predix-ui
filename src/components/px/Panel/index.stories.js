import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { withInfo } from '@storybook/addon-info';
import { withKnobs, text, array, object, boolean, number, select } from '@storybook/addon-knobs';


import Panel from './';

const demoStyles = {
	position: 'relative',
	overflow: 'hidden',
	minHeight: '400px',
	width: '100%',
	boxSizing: 'border-box',
	border: '1px solid gray',
	padding: '1rem'
};

storiesOf('Panel', module)
	.addDecorator(withKnobs)
	.add('default', () => (
		<div style={demoStyles}>
			<p>Use the toggles in properties to invoke the panel.</p>
			<Panel 
				position={select('position', ['top', 'right', 'left', 'bottom'])}
				fullSize={boolean('fullSize', false)}
				minimizable={boolean('minimizable', false)}
				floating={boolean('floating', false)}
				fixed={boolean('fixed', false)}
				persistent={boolean('persistent', false)}
				opened={boolean('opened', true)}>
				{text('children', 'Text Label')}
			</Panel>
		</div>
		
	));
