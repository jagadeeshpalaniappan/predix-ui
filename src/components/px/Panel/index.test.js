import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import Panel from './';

const positions = [ 'left', 'right', 'bottom', 'top' ];
const backgrounds = [ 'light', 'medium', 'dark' ];
describe('Panel', () => {
  test('should render', () =>{
    const wrapper = shallow(
      <Panel position='left'>
        Panel content goes here.
      </Panel>
    );
    expect(wrapper.props().position).toEqual('left');
  });

  describe('backgrounds', () => {
    backgrounds.forEach(background => {
      test(`should render ${background}`, () => {
        const tree = renderer.create(<Panel background={background} opened />).toJSON();
        expect(tree).toMatchSnapshot();
        // expect(tree).toHaveStyleRule('height', 'auto');
      });
    })
  });
  describe('positions', () => {
    positions.forEach(pos => {
      test(`should render ${pos}`, () => {
        const tree = renderer.create(<Panel position={pos} opened />).toJSON();
        expect(tree).toMatchSnapshot();
        // expect(tree).toHaveStyleRule('height', 'auto');
      });
    })
  });

  xtest('renders minimizable', () => {
    const tree = renderer.create(<Panel position='left' minimizable/>).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('width', 'var(--px-panel-size--minimized,4rem)');
  });
  test('renders persistent when opened', () => {
    const tree = renderer.create(<Panel position='left' opened persistent/>).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('position', 'relative');
  });
  test('renders fixed when opened', () => {
    const tree = renderer.create(<Panel position='left' opened fixed/>).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('position', 'fixed');
  });

  test('renders block when opened', () => {
    const tree = renderer.create(<Panel position='left' opened />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('display', 'block');
  });

  test('fixed renders right', () => {
    const tree = renderer.create(<Panel position='right' fixed opened />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('position', 'fixed');
    expect(tree).toHaveStyleRule('right', '0');
  });


  describe('left / right opened', () => {
    test('opened and full size has 100% width', () => {
      const tree = renderer.create(<Panel position='right' opened fullSize fixed/>).toJSON();
      expect(tree).toMatchSnapshot();
      expect(tree).toHaveStyleRule('position', 'fixed');
      expect(tree).toHaveStyleRule('width', '100%');
    });
  });
});
