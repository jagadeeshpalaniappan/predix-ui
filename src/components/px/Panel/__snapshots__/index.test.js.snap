// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel backgrounds should render dark 1`] = `
.c0 {
  overflow: hidden;
  display: block;
  position: absolute;
  z-index: var(--px-panel-z-index,100);
  box-shadow: var(--px-panel-box-shadow,-1px -1px 3px rgba(0,0,0,0.2),1px 1px 3px rgba(0,0,0,0.2));
  background: var(--px-panel-background-color,white);
  border: var(--px-panel-border,none);
  -webkit-transition: var(--px-panel-transition,none);
  transition: var(--px-panel-transition,none);
  background: var(--px-panel-bg-color--dark,#ddd);
  height: auto;
  width: 0;
  top: 0;
  bottom: 0;
  -webkit-transition: var(--px-panel-transition,width 0.4s cubic-bezier(.78,.13,.16,.87));
  transition: var(--px-panel-transition,width 0.4s cubic-bezier(.78,.13,.16,.87));
  white-space: nowrap;
  width: var(--px-panel-size,320px);
  overflow-y: auto;
  right: 0;
  border-left: 1px solid var(--px-panel-border-color,gray);
}

.c1 {
  display: block;
}

<div
  className="c0"
  style={null}
>
  <div
    className="px-panel-content c1"
  />
</div>
`;

exports[`Panel backgrounds should render light 1`] = `
.c0 {
  overflow: hidden;
  display: block;
  position: absolute;
  z-index: var(--px-panel-z-index,100);
  box-shadow: var(--px-panel-box-shadow,-1px -1px 3px rgba(0,0,0,0.2),1px 1px 3px rgba(0,0,0,0.2));
  background: var(--px-panel-background-color,white);
  border: var(--px-panel-border,none);
  -webkit-transition: var(--px-panel-transition,none);
  transition: var(--px-panel-transition,none);
  background: var(--px-panel-bg-color--light,#fff);
  height: auto;
  width: 0;
  top: 0;
  bottom: 0;
  -webkit-transition: var(--px-panel-transition,width 0.4s cubic-bezier(.78,.13,.16,.87));
  transition: var(--px-panel-transition,width 0.4s cubic-bezier(.78,.13,.16,.87));
  white-space: nowrap;
  width: var(--px-panel-size,320px);
  overflow-y: auto;
  right: 0;
  border-left: 1px solid var(--px-panel-border-color,gray);
}

.c1 {
  display: block;
}

<div
  className="c0"
  style={null}
>
  <div
    className="px-panel-content c1"
  />
</div>
`;

exports[`Panel backgrounds should render medium 1`] = `
.c0 {
  overflow: hidden;
  display: block;
  position: absolute;
  z-index: var(--px-panel-z-index,100);
  box-shadow: var(--px-panel-box-shadow,-1px -1px 3px rgba(0,0,0,0.2),1px 1px 3px rgba(0,0,0,0.2));
  background: var(--px-panel-background-color,white);
  border: var(--px-panel-border,none);
  -webkit-transition: var(--px-panel-transition,none);
  transition: var(--px-panel-transition,none);
  background: var(--px-panel-bg-color--medium,#eee);
  height: auto;
  width: 0;
  top: 0;
  bottom: 0;
  -webkit-transition: var(--px-panel-transition,width 0.4s cubic-bezier(.78,.13,.16,.87));
  transition: var(--px-panel-transition,width 0.4s cubic-bezier(.78,.13,.16,.87));
  white-space: nowrap;
  width: var(--px-panel-size,320px);
  overflow-y: auto;
  right: 0;
  border-left: 1px solid var(--px-panel-border-color,gray);
}

.c1 {
  display: block;
}

<div
  className="c0"
  style={null}
>
  <div
    className="px-panel-content c1"
  />
</div>
`;

exports[`Panel fixed renders right 1`] = `
.c0 {
  overflow: hidden;
  display: block;
  position: absolute;
  z-index: var(--px-panel-z-index,100);
  box-shadow: var(--px-panel-box-shadow,-1px -1px 3px rgba(0,0,0,0.2),1px 1px 3px rgba(0,0,0,0.2));
  background: var(--px-panel-background-color,white);
  border: var(--px-panel-border,none);
  -webkit-transition: var(--px-panel-transition,none);
  transition: var(--px-panel-transition,none);
  background: var(--px-panel-bg-color--light,#fff);
  position: fixed;
  height: auto;
  width: 0;
  top: 0;
  bottom: 0;
  -webkit-transition: var(--px-panel-transition,width 0.4s cubic-bezier(.78,.13,.16,.87));
  transition: var(--px-panel-transition,width 0.4s cubic-bezier(.78,.13,.16,.87));
  white-space: nowrap;
  width: var(--px-panel-size,320px);
  overflow-y: auto;
  right: 0;
  border-left: 1px solid var(--px-panel-border-color,gray);
}

.c1 {
  display: block;
}

<div
  className="c0"
  style={null}
>
  <div
    className="px-panel-content c1"
  />
</div>
`;

exports[`Panel left / right opened opened and full size has 100% width 1`] = `
.c0 {
  overflow: hidden;
  display: block;
  position: absolute;
  z-index: var(--px-panel-z-index,100);
  box-shadow: var(--px-panel-box-shadow,-1px -1px 3px rgba(0,0,0,0.2),1px 1px 3px rgba(0,0,0,0.2));
  background: var(--px-panel-background-color,white);
  border: var(--px-panel-border,none);
  -webkit-transition: var(--px-panel-transition,none);
  transition: var(--px-panel-transition,none);
  background: var(--px-panel-bg-color--light,#fff);
  position: fixed;
  height: auto;
  width: 0;
  top: 0;
  bottom: 0;
  -webkit-transition: var(--px-panel-transition,width 0.4s cubic-bezier(.78,.13,.16,.87));
  transition: var(--px-panel-transition,width 0.4s cubic-bezier(.78,.13,.16,.87));
  white-space: nowrap;
  width: var(--px-panel-size,320px);
  overflow-y: auto;
  width: 100%;
  overflow-y: auto;
  right: 0;
  border-left: 1px solid var(--px-panel-border-color,gray);
}

.c1 {
  display: block;
}

<div
  className="c0"
  style={null}
>
  <div
    className="px-panel-content c1"
  />
</div>
`;

exports[`Panel positions should render bottom 1`] = `
.c0 {
  overflow: hidden;
  display: block;
  position: absolute;
  z-index: var(--px-panel-z-index,100);
  box-shadow: var(--px-panel-box-shadow,-1px -1px 3px rgba(0,0,0,0.2),1px 1px 3px rgba(0,0,0,0.2));
  background: var(--px-panel-background-color,white);
  border: var(--px-panel-border,none);
  -webkit-transition: var(--px-panel-transition,none);
  transition: var(--px-panel-transition,none);
  background: var(--px-panel-bg-color--light,#fff);
  width: auto;
  height: 0;
  right: 0;
  left: 0;
  -webkit-transition: var(--px-panel-transition,height 0.4s cubic-bezier(.78,.13,.16,.87));
  transition: var(--px-panel-transition,height 0.4s cubic-bezier(.78,.13,.16,.87));
  height: var(--px-panel-size,320px);
  overflow-y: auto;
  bottom: 0;
  border-top: 1px solid var(--px-panel-border-color,gray);
}

.c1 {
  display: block;
}

<div
  className="c0"
  style={null}
>
  <div
    className="px-panel-content c1"
  />
</div>
`;

exports[`Panel positions should render left 1`] = `
.c0 {
  overflow: hidden;
  display: block;
  position: absolute;
  z-index: var(--px-panel-z-index,100);
  box-shadow: var(--px-panel-box-shadow,-1px -1px 3px rgba(0,0,0,0.2),1px 1px 3px rgba(0,0,0,0.2));
  background: var(--px-panel-background-color,white);
  border: var(--px-panel-border,none);
  -webkit-transition: var(--px-panel-transition,none);
  transition: var(--px-panel-transition,none);
  background: var(--px-panel-bg-color--light,#fff);
  height: auto;
  width: 0;
  top: 0;
  bottom: 0;
  -webkit-transition: var(--px-panel-transition,width 0.4s cubic-bezier(.78,.13,.16,.87));
  transition: var(--px-panel-transition,width 0.4s cubic-bezier(.78,.13,.16,.87));
  white-space: nowrap;
  width: var(--px-panel-size,320px);
  overflow-y: auto;
  left: 0;
  border-right: 1px solid var(--px-panel-border-color,gray);
}

.c1 {
  display: block;
}

<div
  className="c0"
  style={null}
>
  <div
    className="px-panel-content c1"
  />
</div>
`;

exports[`Panel positions should render right 1`] = `
.c0 {
  overflow: hidden;
  display: block;
  position: absolute;
  z-index: var(--px-panel-z-index,100);
  box-shadow: var(--px-panel-box-shadow,-1px -1px 3px rgba(0,0,0,0.2),1px 1px 3px rgba(0,0,0,0.2));
  background: var(--px-panel-background-color,white);
  border: var(--px-panel-border,none);
  -webkit-transition: var(--px-panel-transition,none);
  transition: var(--px-panel-transition,none);
  background: var(--px-panel-bg-color--light,#fff);
  height: auto;
  width: 0;
  top: 0;
  bottom: 0;
  -webkit-transition: var(--px-panel-transition,width 0.4s cubic-bezier(.78,.13,.16,.87));
  transition: var(--px-panel-transition,width 0.4s cubic-bezier(.78,.13,.16,.87));
  white-space: nowrap;
  width: var(--px-panel-size,320px);
  overflow-y: auto;
  right: 0;
  border-left: 1px solid var(--px-panel-border-color,gray);
}

.c1 {
  display: block;
}

<div
  className="c0"
  style={null}
>
  <div
    className="px-panel-content c1"
  />
</div>
`;

exports[`Panel positions should render top 1`] = `
.c0 {
  overflow: hidden;
  display: block;
  position: absolute;
  z-index: var(--px-panel-z-index,100);
  box-shadow: var(--px-panel-box-shadow,-1px -1px 3px rgba(0,0,0,0.2),1px 1px 3px rgba(0,0,0,0.2));
  background: var(--px-panel-background-color,white);
  border: var(--px-panel-border,none);
  -webkit-transition: var(--px-panel-transition,none);
  transition: var(--px-panel-transition,none);
  background: var(--px-panel-bg-color--light,#fff);
  width: auto;
  height: 0;
  right: 0;
  left: 0;
  -webkit-transition: var(--px-panel-transition,height 0.4s cubic-bezier(.78,.13,.16,.87));
  transition: var(--px-panel-transition,height 0.4s cubic-bezier(.78,.13,.16,.87));
  height: var(--px-panel-size,320px);
  overflow-y: auto;
  top: 0;
  border-bottom: 1px solid var(--px-panel-border-color,gray);
}

.c1 {
  display: block;
}

<div
  className="c0"
  style={null}
>
  <div
    className="px-panel-content c1"
  />
</div>
`;

exports[`Panel renders block when opened 1`] = `
.c0 {
  overflow: hidden;
  display: block;
  position: absolute;
  z-index: var(--px-panel-z-index,100);
  box-shadow: var(--px-panel-box-shadow,-1px -1px 3px rgba(0,0,0,0.2),1px 1px 3px rgba(0,0,0,0.2));
  background: var(--px-panel-background-color,white);
  border: var(--px-panel-border,none);
  -webkit-transition: var(--px-panel-transition,none);
  transition: var(--px-panel-transition,none);
  background: var(--px-panel-bg-color--light,#fff);
  height: auto;
  width: 0;
  top: 0;
  bottom: 0;
  -webkit-transition: var(--px-panel-transition,width 0.4s cubic-bezier(.78,.13,.16,.87));
  transition: var(--px-panel-transition,width 0.4s cubic-bezier(.78,.13,.16,.87));
  white-space: nowrap;
  width: var(--px-panel-size,320px);
  overflow-y: auto;
  left: 0;
  border-right: 1px solid var(--px-panel-border-color,gray);
}

.c1 {
  display: block;
}

<div
  className="c0"
  style={null}
>
  <div
    className="px-panel-content c1"
  />
</div>
`;

exports[`Panel renders fixed when opened 1`] = `
.c0 {
  overflow: hidden;
  display: block;
  position: absolute;
  z-index: var(--px-panel-z-index,100);
  box-shadow: var(--px-panel-box-shadow,-1px -1px 3px rgba(0,0,0,0.2),1px 1px 3px rgba(0,0,0,0.2));
  background: var(--px-panel-background-color,white);
  border: var(--px-panel-border,none);
  -webkit-transition: var(--px-panel-transition,none);
  transition: var(--px-panel-transition,none);
  background: var(--px-panel-bg-color--light,#fff);
  position: fixed;
  height: auto;
  width: 0;
  top: 0;
  bottom: 0;
  -webkit-transition: var(--px-panel-transition,width 0.4s cubic-bezier(.78,.13,.16,.87));
  transition: var(--px-panel-transition,width 0.4s cubic-bezier(.78,.13,.16,.87));
  white-space: nowrap;
  width: var(--px-panel-size,320px);
  overflow-y: auto;
  left: 0;
  border-right: 1px solid var(--px-panel-border-color,gray);
}

.c1 {
  display: block;
}

<div
  className="c0"
  style={null}
>
  <div
    className="px-panel-content c1"
  />
</div>
`;

exports[`Panel renders minimizable 1`] = `
.c0 {
  overflow: hidden;
  display: block;
  position: absolute;
  z-index: var(--px-panel-z-index,100);
  box-shadow: var(--px-panel-box-shadow,-1px -1px 3px rgba(0,0,0,0.2),1px 1px 3px rgba(0,0,0,0.2));
  background: var(--px-panel-background-color,white);
  border: var(--px-panel-border,none);
  -webkit-transition: var(--px-panel-transition,none);
  transition: var(--px-panel-transition,none);
  background: var(--px-panel-bg-color--light,#fff);
  width: var(--px-panel-size--minimized,4rem);
  height: auto;
  width: 0;
  top: 0;
  bottom: 0;
  -webkit-transition: var(--px-panel-transition,width 0.4s cubic-bezier(.78,.13,.16,.87));
  transition: var(--px-panel-transition,width 0.4s cubic-bezier(.78,.13,.16,.87));
  white-space: nowrap;
  left: 0;
  border-right: 1px solid var(--px-panel-border-color,gray);
}

.c1 {
  display: block;
}

<div
  className="c0"
  style={null}
>
  <div
    className="px-panel-content c1"
  />
</div>
`;

exports[`Panel renders persistent when opened 1`] = `
.c0 {
  overflow: hidden;
  display: block;
  position: absolute;
  z-index: var(--px-panel-z-index,100);
  box-shadow: var(--px-panel-box-shadow,-1px -1px 3px rgba(0,0,0,0.2),1px 1px 3px rgba(0,0,0,0.2));
  background: var(--px-panel-background-color,white);
  border: var(--px-panel-border,none);
  -webkit-transition: var(--px-panel-transition,none);
  transition: var(--px-panel-transition,none);
  background: var(--px-panel-bg-color--light,#fff);
  position: relative;
  -webkit-flex: 0 1 320px;
  -ms-flex: 0 1 320px;
  flex: 0 1 320px;
  height: auto;
  width: 0;
  top: 0;
  bottom: 0;
  -webkit-transition: var(--px-panel-transition,width 0.4s cubic-bezier(.78,.13,.16,.87));
  transition: var(--px-panel-transition,width 0.4s cubic-bezier(.78,.13,.16,.87));
  white-space: nowrap;
  width: var(--px-panel-size,320px);
  overflow-y: auto;
  left: 0;
  border-right: 1px solid var(--px-panel-border-color,gray);
}

.c1 {
  display: block;
}

<div
  className="c0"
  style={null}
>
  <div
    className="px-panel-content c1"
  />
</div>
`;
